File: plugins/approvej-intellij-plugin/src/test/java/org/approvej/intellij/ApprovedFileUtilPlatformTest.java
package org.approvej.intellij;

import com.intellij.openapi.vfs.VirtualFile;
import com.intellij.testFramework.fixtures.BasePlatformTestCase;

public class ApprovedFileUtilPlatformTest extends BasePlatformTestCase {

  public void testFindReceivedFile() {
    myFixture.addFileToProject("MyTest.byValue-approved.txt", "approved");
    myFixture.addFileToProject("MyTest.byValue-received.txt", "received");
    VirtualFile approved = myFixture.findFileInTempDir("MyTest.byValue-approved.txt");

    VirtualFile result = ApprovedFileUtil.findReceivedFile(approved);

    assertNotNull(result);
    assertEquals("MyTest.byValue-received.txt", result.getName());
  }

  public void testFindReceivedFile_no_extension() {
    myFixture.addFileToProject("MyTest.byValue-approved", "approved");
    myFixture.addFileToProject("MyTest.byValue-received", "received");
    VirtualFile approved = myFixture.findFileInTempDir("MyTest.byValue-approved");

    VirtualFile result = ApprovedFileUtil.findReceivedFile(approved);

    assertNotNull(result);
    assertEquals("MyTest.byValue-received", result.getName());
  }

  public void testFindReceivedFile_with_affix() {
    myFixture.addFileToProject("MyTest.byValue-body-approved.json", "approved");
    myFixture.addFileToProject("MyTest.byValue-body-received.json", "received");
    VirtualFile approved = myFixture.findFileInTempDir("MyTest.byValue-body-approved.json");

    VirtualFile result = ApprovedFileUtil.findReceivedFile(approved);

    assertNotNull(result);
    assertEquals("MyTest.byValue-body-received.json", result.getName());
  }

  public void testFindReceivedFile_no_received_file() {
    myFixture.addFileToProject("MyTest.byValue-approved.txt", "approved");
    VirtualFile approved = myFixture.findFileInTempDir("MyTest.byValue-approved.txt");

    VirtualFile result = ApprovedFileUtil.findReceivedFile(approved);

    assertNull(result);
  }

  public void testFindReceivedFile_non_approved_file() {
    myFixture.addFileToProject("MyTest.java", "class MyTest {}");
    VirtualFile file = myFixture.findFileInTempDir("MyTest.java");

    VirtualFile result = ApprovedFileUtil.findReceivedFile(file);

    assertNull(result);
  }
}

File: plugins/approvej-intellij-plugin/src/test/kotlin/org/approvej/intellij/ApprovedFileUtilPlatformTest.kt
package org.approvej.intellij

import com.intellij.testFramework.fixtures.BasePlatformTestCase

class ApprovedFileUtilPlatformTest : BasePlatformTestCase() {

  fun testFindReceivedFile() {
    myFixture.addFileToProject("MyTest.byValue-approved.txt", "approved")
    myFixture.addFileToProject("MyTest.byValue-received.txt", "received")
    val approved = myFixture.findFileInTempDir("MyTest.byValue-approved.txt")

    val result = ApprovedFileUtil.findReceivedFile(approved)

    assertNotNull(result)
    assertEquals("MyTest.byValue-received.txt", result!!.name)
  }

  fun testFindReceivedFile_no_extension() {
    myFixture.addFileToProject("MyTest.byValue-approved", "approved")
    myFixture.addFileToProject("MyTest.byValue-received", "received")
    val approved = myFixture.findFileInTempDir("MyTest.byValue-approved")

    val result = ApprovedFileUtil.findReceivedFile(approved)

    assertNotNull(result)
    assertEquals("MyTest.byValue-received", result!!.name)
  }

  fun testFindReceivedFile_with_affix() {
    myFixture.addFileToProject("MyTest.byValue-body-approved.json", "approved")
    myFixture.addFileToProject("MyTest.byValue-body-received.json", "received")
    val approved = myFixture.findFileInTempDir("MyTest.byValue-body-approved.json")

    val result = ApprovedFileUtil.findReceivedFile(approved)

    assertNotNull(result)
    assertEquals("MyTest.byValue-body-received.json", result!!.name)
  }

  fun testFindReceivedFile_no_received_file() {
    myFixture.addFileToProject("MyTest.byValue-approved.txt", "approved")
    val approved = myFixture.findFileInTempDir("MyTest.byValue-approved.txt")

    val result = ApprovedFileUtil.findReceivedFile(approved)

    assertNull(result)
  }

  fun testFindReceivedFile_non_approved_file() {
    myFixture.addFileToProject("MyTest.java", "class MyTest {}")
    val file = myFixture.findFileInTempDir("MyTest.java")

    val result = ApprovedFileUtil.findReceivedFile(file)

    assertNull(result)
  }
}
